### ISSUE TITLE
[REVIEW]: fqfa: A pure Python package for genomic sequence files

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2076]
**Submitting author:** @afrubin (<a href="http://orcid.org/0000-0003-1474-605X">Alan F. Rubin</a>)
**Repository:** <a href="https://github.com/CountESS-Project/fqfa/" target ="_blank">https://github.com/CountESS-Project/fqfa/</a>
**Version:** v1.0.0 
**Editor:** @lpantano
**Reviewer:** @natir, @FlorianThibord
**Archive:** <a href="https://doi.org/10.5281/zenodo.3688838" target="_blank">10.5281/zenodo.3688838</a>

## Status

[![status](https://joss.theoj.org/papers/88fbeca561fbdc3b848dad77c688fbce/status.svg)](https://joss.theoj.org/papers/88fbeca561fbdc3b848dad77c688fbce)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/88fbeca561fbdc3b848dad77c688fbce"><img src="https://joss.theoj.org/papers/88fbeca561fbdc3b848dad77c688fbce/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/88fbeca561fbdc3b848dad77c688fbce/status.svg)](https://joss.theoj.org/papers/88fbeca561fbdc3b848dad77c688fbce)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/CountESS-Project/fqfa/" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@natir & @FlorianThibord, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @lpantano know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/CountESS-Project/fqfa/">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@afrubin) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/CountESS-Project/fqfa/">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@afrubin) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @natir, @FlorianThibord it looks like you're currently assigned to review this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
PDF failed to compile for issue #2076 with the following error: 

 /app/vendor/ruby-2.4.4/lib/ruby/2.4.0/find.rb:43:in `block in find': No such file or directory - tmp/2076 (Errno::ENOENT)
	from /app/vendor/ruby-2.4.4/lib/ruby/2.4.0/find.rb:43:in `collect!'
	from /app/vendor/ruby-2.4.4/lib/ruby/2.4.0/find.rb:43:in `find'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-9847f98e9ec6/lib/whedon/processor.rb:61:in `find_paper_paths'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-9847f98e9ec6/bin/whedon:50:in `prepare'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	from /app/vendor/bundle/ruby/2.4.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	from /app/vendor/bundle/ruby/2.4.0/bundler/gems/whedon-9847f98e9ec6/bin/whedon:116:in `<top (required)>'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `load'
	from /app/vendor/bundle/ruby/2.4.0/bin/whedon:23:in `<main>'

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.7287/peerj.preprints.970v1 is OK
- 10.1093/bioinformatics/btp163 is OK
- 10.1093/nar/gkp1137 is OK
- 10.1073/pnas.85.8.2444 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by natir** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02076/joss.02076/10.21105.joss.02076.pdf)

**Comment by natir** [comment]
Hello @lpantano I having some little trouble with the contents of this paper. If I clearly understand the openjournals review process I need to create issue in the paper's repository. 

Sorry It's the first time I review for openjournals

**Comment by afrubin** [comment]
Thanks to @natir I've added a comparison of fqfa and pyfastx (the other currently-maintained FASTQ/FASTA Python module I could find) and included this result in the paper.

**Comment by afrubin** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02076/joss.02076/10.21105.joss.02076.pdf)

**Comment by afrubin** [comment]
I've clarified the documentation and added a reference manuscript thanks to @FlorianThibord.

**Comment by afrubin** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
PDF failed to compile for issue #2076 with the following error: 

 Error reading bibliography ./paper.bib (line 113, column 9):
unexpected "m"
expecting space, ",", white space or "}"
Error running filter pandoc-citeproc:
Filter returned error status 1
Looks like we failed to compile the PDF

**Comment by afrubin** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.02076/joss.02076/10.21105.joss.02076.pdf)

**Comment by afrubin** [comment]
@lpantano I've resolved the issues opened by both reviewers. Are we able to move on to the next step?

**Comment by lpantano** [comment]
@natir, can you let me know if all is good on your side? I see some items in the list unchecked. Thanks!

**Comment by natir** [comment]
Yes all it's good for me.

**Comment by lpantano** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.7287/peerj.preprints.970v1 is OK
- 10.1093/bioinformatics/btp163 is OK
- 10.1093/nar/gkp1137 is OK
- 10.1073/pnas.85.8.2444 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by lpantano** [comment]
@afrubin, can you make a zenodo archive of your repo, and make sure title and authors match the paper information? After that we can move forward to acceptance.

**Comment by afrubin** [comment]
@lpantano I have created a new release with zenodo archive that matches the paper information here: https://zenodo.org/record/3688838

I've also updated the package in pypi: https://pypi.org/project/fqfa/1.0.0/

**Comment by lpantano** [comment]
@whedon set v1.0.0  as version

**Comment by whedon** [comment]
OK. v1.0.0  is the version.

**Comment by lpantano** [comment]
@whedon set 10.5281/zenodo.3688838 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.3688838" target="_blank">10.5281/zenodo.3688838</a> is the archive.

**Comment by lpantano** [comment]
@@openjournals/joss-eics, I think this is good to go.

**Comment by labarba** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.7287/peerj.preprints.970v1 is OK
- 10.1093/bioinformatics/btp163 is OK
- 10.1093/nar/gkp1137 is OK
- 10.1073/pnas.85.8.2444 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1342

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1342, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by labarba** [comment]
Hi folks —  I am the Associate Editor-in-Chief on rotation this week. I'm going to send this back to you for some more work, because the paper does not meet [our expectations](https://joss.readthedocs.io/en/latest/submitting.html#what-should-my-paper-contain). This paper is very short: the [word counter](https://wordcounter.net) tool says 197 words. While we don't expect a full article, we ask for 250–1000 words. Our docs highlight this minimum content:

- the summary describing high-level functionality and purpose of the software for a diverse, non-specialist audience;
- a clear statement of need that illustrates the research application of the software (including references to other software meeting similar needs).

TL;DR: the article needs to be improved with more description of how the software works and how it is used in research. 

Thanks!

**Comment by afrubin** [comment]
@labarba Thank you for this feedback. When I did the word count myself, I had included the acknowledgements and counted from the rendered PDF rather than the source.

I have added extra detail throughout the text and should now meet the word count requirements.

The updated zenodo record is here: https://zenodo.org/record/3690230

**Comment by labarba** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary:

OK DOIs

- 10.7287/peerj.preprints.970v1 is OK
- 10.1093/bioinformatics/btp163 is OK
- 10.1093/nar/gkp1137 is OK
- 10.1073/pnas.85.8.2444 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published

. Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1343

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1343, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by labarba** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1234255074043580418) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/1344
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.02076](https://doi.org/10.21105/joss.02076)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? notify your editorial technical team...

**Comment by labarba** [comment]
Congratulations, @afrubin, your JOSS paper is now published! 🚀 

Big thanks to our editor: @lpantano, and the reviewers: @natir, @FlorianThibord — we couldn't do this without you 🙏

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02076/status.svg)](https://doi.org/10.21105/joss.02076)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02076">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02076/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02076/status.svg
   :target: https://doi.org/10.21105/joss.02076
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02076/status.svg)](https://doi.org/10.21105/joss.02076)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
